Tasks/VsTest/VSTest.ts

```diff
@@ -0,0 +1,501 @@
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+
+import tl = require('vsts-task-lib/task');
+var os = require('os');
+import path = require('path');
+import Q = require('q');
+var regedit = require('regedit');
+var uuid = require('node-uuid');
+var fs = require('fs');
+var xml2js = require('xml2js');
+
+try{
+    var vsTestVersion: string = tl.getInput('vsTestVersion');
+    var testAssembly: string = tl.getInput('testAssembly', true);
+    var testFiltercriteria: string = tl.getInput('testFiltercriteria');
+    var runSettingsFile: string = tl.getPathInput('runSettingsFile');
+    var codeCoverageEnabled: boolean = tl.getBoolInput('codeCoverageEnabled');
+    var pathtoCustomTestAdapters: string = tl.getInput('pathtoCustomTestAdapters');
+    var overrideTestrunParameters: string = tl.getInput('overrideTestrunParameters');
+    var otherConsoleOptions: string = tl.getInput('otherConsoleOptions');
+    var testRunTitle: string = tl.getInput('testRunTitle');
+    var platform: string = tl.getInput('platform');
+    var configuration: string = tl.getInput('configuration');
+    var publishRunAttachments: string = tl.getInput('publishRunAttachments');
+    var runInParallel: boolean = tl.getBoolInput('runInParallel');
+    
+    //Write-Host "##vso[task.logissue type=warning;TaskName=VSTest]"
+
+    var sourcesDirectory = tl.getVariable('System.DefaultWorkingDirectory');
+    var testAssemblyFiles = [];
+    if(testAssembly.indexOf('*') >= 0 || testAssembly.indexOf('?') >= 0) {
+        tl.debug('Pattern found in solution parameter.');
+        var excludeTestAssemblies = [];
+        var allFiles = tl.find(sourcesDirectory);
+        var testAssemblyFilters = testAssembly.split(';');
+        testAssemblyFilters.forEach(function(testAssemblyFilter){
+            if(testAssemblyFilter.startsWith("-:")){
+                if(testAssemblyFilter.indexOf('*') >= 0 || testAssemblyFilter.indexOf('?') >= 0){                    
+                    excludeTestAssemblies = excludeTestAssemblies.concat(getFilteredFiles(testAssemblyFilter.substr(2), allFiles));
+                }
+                else{
+                    excludeTestAssemblies.push(testAssemblyFilter.substr(2));
+                }
+            }
+            else if(testAssemblyFilter.indexOf('*') >= 0 || testAssemblyFilter.indexOf('?') >= 0){
+                testAssemblyFiles = testAssemblyFiles.concat(getFilteredFiles(testAssemblyFilter, allFiles));
+            }
+            else{
+                testAssemblyFiles.push(testAssemblyFilter);
+            }
+        });
+        testAssemblyFiles = removeArrayFromArray(testAssemblyFiles, excludeTestAssemblies);              
+    }
+    else{
+        tl.debug('No Pattern found in solution parameter.');
+        testAssembly.replace(';;', "`0")  // Borrowed from Legacy File Handler
```

Comment:
Don't do that. that logic is written for Find-Files functionality which is pure PS/C# thing.